Update, and add recipes to build FairRoot/FairSoft with AliBuild

MohammadAlTurany opened the pull request.

dennisklein commented on boost.sh:
```diff
@@ -1,18 +1,18 @@
 version: "%(tag_basename)s%(defaults_upper)s"
 source: https://github.com/alisw/boost.git
-tag: v1.59.0
+tag: v1.64.0
```
It is installed from alisw repo. Make sure they have updated it to recent boost.

rbx replied:
Ah yeah. I will make an issue there if they can update it.

fuhlig1 commented:
Fixed with 2bf5f644e57496a2c72fc39fa228ed37696575f8

rbx commented:
Does it make more sense to try to submit this kind of change to alidist directly? Project like O2 uses FairMQ, but also uses alidist, not alfadist...

dennisklein commented:
Well, they will need to update, if FairRoot changes its deps. @fuhlig1 the recpie you merged will not work at the moment, because the alisw/boost repo does not contain version 1.64.

Commit:
Add Support for G4PY, XercesC and add automated version detection for GEANT4 data files

sawenzel committed:
Generalize G4 recipe; Upgrade sim environment for O2

This is a complex commit with the goal to bump the versions
of the simulation infrastructure for O2. In order to do so, the
following things are done here:

 * make the Geant4 recipe autodiscover the versions installed
   data files
   (previously they were hard-coded, preventing to override
    the package in defaults)

MohammadAlTurany committed:
update the README

MohammadAlTurany committed:
1- Remove alice specific recipes
2- Update the README
3- Do not run the tests when building ZMQ

MohammadAlTurany committed:
Remove un-used recipes

MohammadAlTurany committed:
boost 1.66

MohammadAlTurany committed:
remove un-needed dependencies

MohammadAlTurany committed:
cleaning recipes

MohammadAlTurany committed:
Bump FairMq to 1.2.0 and small correction in README

MohammadAlTurany committed:
Bump Boost, Geant4, root, Gean4_vmc
Change location for Geant3 and Boost

MohammadAlTurany committed:
Bump FairMq to 1.2.1

MohammadAlTurany committed:
typo

MohammadAlTurany committed:
correct source and Tag of G4, adapt the g4py recipe to alfadist

MohammadAlTurany committed:
correct source for Xercesc and remove openGL denpendancy

MohammadAlTurany committed:
add new recipe for ICU

MohammadAlTurany committed:
G4PY
Fork Geant4 to FairRootGroup
Apply patch to solve the problem with boost_python library name for boost 1.67

MohammadAlTurany committed:
Build boost-python libraries

MohammadAlTurany committed:
do not run the g4py tests
